src/components/TableDialog.vue

```diff
@@ -111,130 +111,125 @@
 </template>
 
 <script lang="ts">
-import Vue, { PropType } from 'vue';
+import {
+  defineComponent, ref, Ref, computed,
+} from '@vue/composition-api';
 
 import api from '@/api';
 import { FileType, CSVColumnType } from '@/types';
 import { validFileType, fileName as getFileName, csvFileTypeRecommendations } from '@/utils/files';
 
 const defaultKeyField = '_key';
-export default Vue.extend({
-  name: 'TableDialog',
+const fileTypes: readonly FileType[] = [
+  {
+    extension: ['csv'],
+    queryCall: 'csv',
+    hint: 'Comma Separated Value file',
+    displayName: 'CSV',
+  },
+];
 
+export default defineComponent({
+  name: 'TableDialog',
   props: {
     workspace: {
-      type: String as PropType<string>,
+      type: String,
       required: true,
     },
   },
-  data() {
-    return {
-      tableDialog: false,
-      types: [
-        {
-          extension: ['csv'],
-          queryCall: 'csv',
-          hint: 'Comma Separated Value file',
-          displayName: 'CSV',
-        },
-      ] as FileType[],
-      file: null as File | null,
-      fileName: null as string | null,
-      fileUploadError: null as string | null,
-      tableCreationError: null as string | null,
-      key: defaultKeyField,
-      overwrite: false,
-      columnType: {} as { [key: string]: CSVColumnType },
-      uploading: false,
-      uploadProgress: null as number | null,
-    };
-  },
-  computed: {
-    createDisabled(): boolean {
-      return (
-        !this.file
-        || !this.fileName
-        || !!this.fileUploadError
-      );
-    },
-
-    multinetTypes(): CSVColumnType[] {
-      return [
-        'label',
-        'category',
-        'number',
-        'date',
-      ];
-    },
-  },
-
-  methods: {
-    restoreKeyField() {
-      this.key = defaultKeyField;
-    },
-
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    async handleFileInput(this: any, file: File | undefined) {
-      this.file = file || null;
+  setup(props, { emit }) {
+    // Type recommendation
+    const columnType: Ref<{[key: string]: CSVColumnType}> = ref({});
+    const multinetTypes: readonly CSVColumnType[] = ['label', 'category', 'number', 'date'];
```

Comment:
I wonder if this should be hoisted to the top level. (Totally fine to leave it here now, I'm really just wondering.)

Reply:
This is the only question I had, too. Is it just reactive things that should be set inside the `setup()` function and everything static should be moved into the top level (outside of the `defineComponent`)?

Reply:
There's no rule about non-reactive things not being declared inside `setup()`. In fact, in general, you may want to track some values only during setup, and not expose them to the outside.

In this case, maintaining a list of semantic types that the application knows about might make sense. But since right now this is the only component that cares, we could leave it like this as well.

Reply:
You're right, it should be outside of setup. I originally considered placing it outside of setup, but incorrectly thought since I was returning it to the template, that it should be in setup (don't know why I was thinking that tbh). I'll fix it.